### ISSUE TITLE
How Flash attention2 use in Prefix decoder?

### ISSUE THREAD

**Issue by lonelycrab888** [Dao-AILab/flash-attention#682]
My attention_mask is a dynamic mask matrix for the prefix decoder, similar to UniLM and GLM. How should this type of attention_mask be applied to Flash Attention?

**Comment by tridao** [comment]
That kind of mask is not currently supported.
